client/v2/broadcast/factory.go

```diff
@@ -0,0 +1,41 @@
+package broadcast
+
+import (
+	"context"
+	"fmt"
+
+	"cosmossdk.io/client/v2/broadcast/types"
+	"cosmossdk.io/client/v2/internal/comet"
+)
+
+var _ types.BroadcasterFactory = &Factory{}
+
+// Factory is a factory for creating Broadcaster instances.
+type Factory struct {
+	engines map[string]types.NewBroadcasterFn
+}
+
+// Create creates a new Broadcaster based on the given consensus type.
+func (f *Factory) Create(_ context.Context, consensus, url string, opts ...types.Option) (types.Broadcaster, error) {
```

Comment:
Why does it take a context?

Reply:
nice catch, `NewBroadcasterFn` should take that context